src/models/vec_model.jl

```diff
@@ -57,33 +65,55 @@ function set_objective_multiple!(model::VecModel, m)
     return model
 end
 
-function optimize(model::VecModel, optimizer::AbstractOptimizer, x0, args...; kwargs...)
-    workspace = Workspace(model, optimizer, copy(x0), args...; kwargs...)
+"""
+Generic `optimize` for VecModel
+"""
+function optimize(model::VecModel, optimizer::AbstractOptimizer, x0::Nothing, args...; options=nothing, kwargs...)
+    if options isa Nothing
+        workspace = Workspace(model, optimizer, args...; kwargs...)
+    else
+        workspace = Workspace(model, optimizer, args...; options=options, kwargs...)
+    end
     return optimize!(workspace)
 end
 
-"""
- Workspace constructor without x0
-"""
-function optimize(model::VecModel, optimizer::AbstractOptimizer, args...; kwargs...)
-    workspace = Workspace(model, optimizer, args...; kwargs...)
+function optimize(model::VecModel, optimizer::AbstractOptimizer, x0, args...; options=nothing, kwargs...)
+    if options isa Nothing
+        workspace = Workspace(model, optimizer, copy(x0), args...; kwargs...)
+    else
+        workspace = Workspace(model, optimizer, copy(x0), args...; options=options, kwargs...)
+    end
     return optimize!(workspace)
 end
 
-function tovecmodel(m::AbstractModel, x0 = getmin(m))
+function tovecmodel(m::AbstractModel, x0=nothing)
+    if x0 isa Nothing
+        x0 = getmin(m)
+    end
     v, _unflatten = flatten(x0)
     unflatten = Unflatten(x0, _unflatten)
     return VecModel(
+        # objective
         Objective(x -> m.objective(unflatten(x)), flags = m.objective.flags),
+        # eq_constraints
         length(m.eq_constraints.fs) != 0 ? VectorOfFunctions(map(m.eq_constraints.fs) do c
             EqConstraint(x -> maybeflatten(c.f(unflatten(x)))[1], maybeflatten(c.rhs)[1], c.dim, c.flags)
         end) : VectorOfFunctions(EqConstraint[]),
+        # ineq_constraints
         length(m.ineq_constraints.fs) != 0 ? VectorOfFunctions(map(m.ineq_constraints.fs) do c
             IneqConstraint(x -> maybeflatten(c.f(unflatten(x)))[1], maybeflatten(c.rhs)[1], c.dim, c.flags)
         end) : VectorOfFunctions(IneqConstraint[]),
+        # sd_constraints
+        length(m.sd_constraints.fs) != 0 ? VectorOfFunctions(map(m.ineq_constraints.fs) do c
+            SDConstraint(x -> maybeflatten(c.f(unflatten(x)))[1], c.dim)
```

Comment:
Remove `maybeflatten` here. We want the output to be a matrix not a vector, This is used to vectorize the output in the inequality and equality constraints.

Reply:
also you are mapping `m.ineq_constraints` here not `m.sd_constraints`